File: java/Swing.java
import javax.swing.*;

public class Swing {
    private static void initComponents() {
        JFrame frame = new JFrame("Hello World Swing");
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

        JLabel label = new JLabel("Hello World");
        frame.getContentPane().add(label);

        frame.pack();
        frame.setSize(300, 100);
        frame.setVisible(true);
    }

    public static void main(String[] args) {
        javax.swing.SwingUtilities.invokeLater(new Runnable() {
            public void run() {
                initComponents();
            }
        });
    }
}

File: scala/swing.scala
import javax.swing.JFrame
import javax.swing.JLabel

val frame = new JFrame("Hello World Swing")
frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE)

val label = new JLabel("Hello World")
frame.getContentPane().add(label)

frame.pack()
frame.setSize(300, 100)
frame.setVisible(true)
